### PR TITLE
2880: Fix repeating dates. Again.

### PR DESCRIPTION
**Pull request by LeandraH**
### Short description

The repeating dates were off again. Reminder, the underlying problem is that the package `rrule` handles time zones by ignoring them and pretending that everything is in UTC. This time, we had the problem that e.g. all-day events arrive with a starting time of 0:00+2 which translates to 22:00 on the previous day in UTC. If we then go and get for example the next Saturday from that, we get Saturday at 22:00 in UTC, which translates to midnight on Sunday in local time.

Links for testing: http://localhost:9000/testumgebung/de/events/jeden-donnerstag and https://cms-test.integreat-app.de/testumgebung/events/de/4596/edit/

Note: I haven't found a well-supported alternative package, but man, would I be happy to switch.

### Proposed changes

- Move the start date in the recurrence rule to local time, then get the next recurrences, then set those back to UTC time
- Added a couple of more tests for different situations I could think of

### Side effects

- It should be none. I decided against keeping the offset on the recurrences to keep them in the existing (and frankly, correct) date format.

### Resolved issues

Fixes: #2880

### CODE REVIEW

File: shared/api/models/DateModel.ts [hunk 1]
```diff
@@ -128,23 +138,16 @@ class DateModel {
...
-  private currentDateToRrule(): Date {
-    // rrule is not correctly handling timezones and is expecting dates in "local time", i.e. in the timezone of dstart
-    // Therefore we have to manually apply timezone offset, e.g. for daylight savings time
-    // https://github.com/jkbrzt/rrule#important-use-utc-dates
-    // Also include dates happening at the moment by subtracting the duration
-    return DateTime.now()
-      .minus(this._duration)
-      .minus({ minutes: this._offset - DateTime.now().offset })
-      .toJSDate()
+  private getRecurrenceRuleInLocalTime(recurrenceRule: RRuleType): RRuleType | undefined {
```
Comment (steffenkleinle) [inline shared/api/models/DateModel.ts:141]: :wrench: No need for `undefined`. Please also remove check above.
```suggestion
  private getRecurrenceRuleInLocalTime(recurrenceRule: RRuleType): RRuleType {
```

**Review by steffenkleinle — APPROVED** [review]
Looks a lot simpler than before, great job! I don't understand everything happening and why it works, but looks good and seems to work! Great job :tada:

File: shared/api/models/DateModel.ts (hunk 1 above)
Comment (LeandraH) [reply shared/api/models/DateModel.ts:141]: Good catch, thanks! Too much refactoring 🤪

**Review by f1sh1918 — APPROVED** [review]
nicely done 👍

**Pull request merged**